frontend/src/utility_components/sidebar/Sidebar.tsx

```diff
@@ -0,0 +1,78 @@
+import { useState } from 'react'
+import './sidebar.css'
+import logo from '../../images/edited/AEYB_A0_Circle_resized.png'
+import Menuitem from './menuitem/Menuitem'
+import { Outlet } from 'react-router-dom';
+import { FontAwesomeIconProps } from '@fortawesome/react-fontawesome';
+import { faCalendar, faHouse, faUser } from '@fortawesome/free-solid-svg-icons';
+import LogoutButton from '../LogoutButton';
+import { useUserContext } from '../../contexts/UserContext';
+
+export interface MenuItemData {
+    url: string;
+    icon: FontAwesomeIconProps["icon"],
+    title: string;
+}
+
+const MenuItems: MenuItemData[] = [{
+    url: '/',
+    icon: faHouse,
+    title: 'HOME',
+}, {
+    url: '/calendarpage',
+    icon: faCalendar,
+    title: 'CALENDAR',
+}, {
+    url: '/profilepage',
+    icon: faUser,
+    title: 'PROFILE'
+}];
+
+export default function Sidebar() {
+
+    const [currentPage, setCurrentPage] = useState('HOME')
```

Comment:
I'm just thinking and I believe a more robust way this could be done is like:
```ts
const location = useLocation();

// When you're checking for if a page is the current page:
isCurrentPage ={page.url === location.pathname}
```
This will remove the need for a `setCurrentPage` callback in `Menuitem` and should fix the issue when you navigate to a page that's not one of the 3 options here.

For more info on `useLocation` see: https://reactgo.com/react-router-current-route/